ark/phenotyping/som_utils_test.py

```diff
@@ -11,115 +12,241 @@
 import ark.utils.test_utils as test_utils
 
 
-def mocked_cluster_pixels(base_dir, chan_list):
-    # assign marker names and create sample marker information
-    sample_pixel_mat = pd.read_csv(os.path.join(base_dir, 'pixel_mat_preprocessed.csv'),
-                                   usecols=chan_list)
+def mocked_train_som(fovs, channels, base_dir,
+                     subset_name='pixel_mat_subsetted.hdf5', weights_name='weights.hdf5'):
+    # define the matrix we'll be training on
+    pixel_mat_sub = pd.DataFrame(columns=channels)
 
-    sample_pixel_mat = sample_pixel_mat
+    for fov in fovs:
+        # read the specific fov from the subsetted HDF5
+        fov_mat_sub = pd.read_hdf(os.path.join(base_dir, subset_name), key=fov)
 
-    # we will assign cluster labels based on the average of the rows
-    sample_pixel_mat['Cluster_ID'] = sample_pixel_mat.mean(axis=1)
+        # only take the channel columns
+        fov_mat_sub = fov_mat_sub[channels]
 
-    # multiply by 100 and truncate for an integer cluster assignment
-    sample_pixel_mat['Cluster_ID'] *= 100
-    sample_pixel_mat['Cluster_ID'] = sample_pixel_mat['Cluster_ID'].astype(int)
+        # append to pixel_mat_sub
+        pixel_mat_sub = pd.concat([pixel_mat_sub, fov_mat_sub])
 
-    # write the final data to CSV
-    sample_pixel_mat.to_csv(os.path.join(base_dir, 'pixel_mat_clustered.csv'), index=False)
+    # FlowSOM flattens the weights dimensions, ex. 10x10x10 becomes 100x10
+    weights = np.random.rand(100, len(channels))
+
+    # take 100 random rows from pixel_mat_sub, and element-wise multiply weights by that
+    multiply_factor = pixel_mat_sub.sample(n=100).values
+    weights = weights * multiply_factor
+
+    # write weights to HDF5
+    with h5py.File(os.path.join(base_dir, weights_name), 'w') as hf:
+        hf.create_dataset('weights', data=weights)
+
+
+def mocked_cluster_pixels(fovs, channels, base_dir,
+                          pre_name='pixel_mat_preprocessed.hdf5',
+                          weights_name='weights.hdf5',
+                          cluster_name='pixel_mat_clustered.hdf5'):
+    # read in the weights matrix
+    with h5py.File(os.path.join(base_dir, weights_name), 'r') as hf:
+        weights = hf['weights'][:]
```

Comment:
Lets do some operation with the weights to make sure they're not corrupted, and are the correct shape